python/ray/tune/registry.py

```diff
@@ -179,8 +198,29 @@ def _make_key(prefix: str, category: str, key: str):
 
 class _Registry:
     def __init__(self, prefix: Optional[str] = None):
+        """If no prefix is given, use runtime context job ID."""
         self._to_flush = {}
-        self._prefix = prefix or uuid.uuid4().hex[:8]
+        self._prefix = prefix
+        self._registered = set()
+        self._atexit_handler_registered = False
+
+    @property
+    def prefix(self):
+        if not self._prefix:
+            self._prefix = ray.get_runtime_context().get_job_id()
+        return self._prefix
+
+    def _register_atexit(self):
+        if self._atexit_handler_registered:
+            # Already registered
+            return
+
+        if ray._private.worker.global_worker.mode != ray.SCRIPT_MODE:
+            # Only cleanup on the driver
```

Comment:
I think it's acceptable to not care about Ray client here. The garbage we may collect should be small in any case, and we didn't do any cleanup previously at all.

Reply:
That seems ok, though wouldn't this also work in client mode automatically, as the Tune driver is usually run on the driver in client mode too?

Reply:
The default behavior is to wrap the whole tune run/tuner object in a remote task/actor, which would then be WORKER_MODE